### PR TITLE
Query config

### PR DESCRIPTION
**Pull request by tiagovtristao**
Add a new `plz query config` subcommand, allowing human readable or json format to be printed.

`plz query config`:
```
[please]      
version = 16.14.0             
toolsurl =           
location = /home/ttristao/code/please/plz-out/please
selfupdate = true                    
downloadlocation = https://get.please.build
numoldversions = 10                   
autoclean = true                         
numthreads = 14                          
defaultrepo =                                 
                          
[parse]                 
buildfilename = BUILD                
buildfilename = BUILD.plz       
blacklistdirs = tree               
builddefsdir = build_defs
numthreads = 14                   
gitfunctions = true 
...
```

`plz query config parse.buildfilename`:
```
BUILD
BUILD.plz
```

`plz query config --json`:
```
{                        
    "please": {                                            
        "version": "16.14.0",                            
        "toolsurl": "",                                      
        "versionchecksum": null,
        "location": "/home/ttristao/code/please/plz-out/please",                         
        "selfupdate": true,                                                             
        "downloadlocation": "https://get.please.build",
        "numoldversions": 10,                        
        "autoclean": true,                     
        "numthreads": 14,                
        "motd": null,               
        "defaultrepo": ""                                   
    },                                                                                
    "parse": {                                                        
        "experimentaldir": null,       
        "buildfilename": [                     
            "BUILD",                 
            "BUILD.plz"              
        ],                              
        "blacklistdirs": [              
            "tree"               
        ],                               
        "preloadbuilddefs": null,         
        "preloadsubincludes": null,   
        "builddefsdir": [        
            "build_defs"                 
        ],                                
        "numthreads": 14,                              
        "gitfunctions": true       
    }, 
   ...
}
```

### CODE REVIEW

File: test/build_defs/test.build_defs [hunk 1]
```diff
@@ -23,13 +23,13 @@ def please_repo_e2e_test(
     ]
 
     if expected_output:
-       test_cmd += [f"$TOOLS_CONTENT_CHECKER '{o}' '{c}'" for o, c in expected_output.items()]
+        test_cmd += [f"$TOOLS_CONTENT_CHECKER '{o}' '{c}'" for o, c in expected_output.items()]
```
Comment (tiagovtristao) [inline test/build_defs/test.build_defs:26]: Not the most elegant but gets the job done in a POSIX compliant way. This change was required to make newlines in multiline strings not to cause grep to return false positives.

**Review by samwestmoreland — APPROVED** [review]

**Review by Tatskaari — APPROVED** [review]
Looks good to me :)

File: src/query/config.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+// Config prints configuration settings in human-readable format.
+func Config(config *core.Configuration, options []string) {
+	if len(options) == 0 {
+		v, err := gcfg.Stringify(config)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Print(v)
```
[the remark this answers is not in the document]
Comment (tiagovtristao) [reply src/query/config.go:21]: The output already comes with a new line.

File: src/please.go [hunk 3]
```diff
@@ -414,7 +414,11 @@ var opts struct {
 		} `command:"filter" description:"Filter the given set of targets according to some rules"`
 		RepoRoot struct {
 		} `command:"reporoot" alias:"repo_root" description:"Output the root of the current Please repo"`
-	} `command:"query" description:"Queries information about the build graph"`
+		Config struct {
+			Options []string `long:"option" description:"Print specific option."`
```
[the remark this answers is not in the document]
Comment (tiagovtristao) [reply src/please.go:418]: I was trying to keep it consistent with with other commands. i.e. `plz query printf --field`

**Pull request merged**